### PR TITLE
✨Computational backend: passing labels and envs 🗃️

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
- [x] sets user_id, study_id, uuid (node_id), product_name, simcore_user_agent on computational service containers (allows to filter puppeteer runs from real users for monitoring purpose @elisabettai , allows resource tracking)
- [x] virtually allow to pass any kind of run_metadata on the computational service containers
- [x] passes environment variables to computational service containers, they are now also contained in *comp_tasks* table
- [x] 🗃️ new metadata column on *comp_runs* table, necessary to pass container labels
- [x] substitutes osparc variables thanks to @pcrespov

@matusdrobuliak66 the limits will come in a different PR: https://github.com/ITISFoundation/osparc-simcore/pull/4453
I need to do some changes to the naming of these labels because ```_``` are not valid docker label keys. I will have to change these.

## Related issue/s

- resolves https://github.com/ITISFoundation/osparc-simcore/issues/4350
- resolves https://github.com/ITISFoundation/osparc-simcore/issues/4271
- resolves https://github.com/ITISFoundation/osparc-simcore/issues/3908

### CODE REVIEW

File: packages/models-library/src/models_library/service_settings_labels.py [hunk 1]
```diff
@@ -490,7 +490,7 @@ class SimcoreServiceLabels(DynamicSidecarServiceLabels):
     spec will be generated before starting the service.
     """
 
-    settings: Json[SimcoreServiceSettingsLabel] = Field(
+    settings: Json[SimcoreServiceSettingsLabel] | None = Field(
```
Comment (pcrespov) [inline packages/models-library/src/models_library/service_settings_labels.py:493]: I would avoid this change if possible. You made it nullable. Is that what youwanted to make or rather not-required?

File: services/dask-sidecar/src/simcore_service_dask_sidecar/computational_sidecar/docker_utils.py [hunk 2]
```diff
@@ -44,33 +51,40 @@
...
     comp_volume_mount_point: str,
     boot_mode: BootMode,
     task_max_resources: dict[str, Any],
+    envs: ContainerEnvsDict,
+    labels: ContainerLabelsDict,
 ) -> DockerContainerConfig:
     nano_cpus_limit = int(task_max_resources.get("CPU", 1) * 1e9)
     memory_limit = ByteSize(task_max_resources.get("RAM", 1024**3))
+    env_variables = [
+        f"{name.upper()}_FOLDER=/{name}s"
```
Comment (pcrespov) [inline services/dask-sidecar/src/simcore_service_dask_sidecar/computational_sidecar/docker_utils.py:68]: MINOR: for three items i find more clear to write them exclicitly than a for loop :-)

**Review by pcrespov — APPROVED** [review]
:tada: !

File: packages/models-library/src/models_library/service_settings_labels.py (hunk 1 above)
Comment (sanderegg) [reply packages/models-library/src/models_library/service_settings_labels.py:493]: there are old services, such as the sleeper 1.0.0 that do not have any label set.

File: services/dask-sidecar/src/simcore_service_dask_sidecar/computational_sidecar/models.py [hunk 3]
```diff
@@ -42,7 +40,8 @@ def ensure_no_memory_swap_means_no_swap(cls, v, values):
     @classmethod
     def ensure_memory_swap_cannot_be_unlimited_nor_smaller_than_memory(cls, v, values):
         if v < values["memory"]:
-            raise ValueError("Memory swap cannot be set to a smaller value than memory")
+            msg = "Memory swap cannot be set to a smaller value than memory"
```
Comment (GitHK) [inline services/dask-sidecar/src/simcore_service_dask_sidecar/computational_sidecar/models.py:43]: I see we want to enforce this pattern now. I will try remember it.

File: services/web/server/src/simcore_service_webserver/director_v2/_handlers.py [hunk 4]
```diff
@@ -56,11 +60,16 @@ async def start_computation(request: web.Request) -> web.Response:
         force_restart = bool(body.get("force_restart", force_restart))
         cluster_id = body.get("cluster_id")
 
+    simcore_user_agent = request.headers.get(
+        X_SIMCORE_USER_AGENT, UNDEFINED_DEFAULT_SIMCORE_USER_AGENT_VALUE
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/director_v2/_handlers.py:64]: 👍

**Review by GitHK — APPROVED** [review]
👍 nice

File: packages/models-library/src/models_library/service_settings_labels.py (hunk 1 above)
Comment (sanderegg) [reply packages/models-library/src/models_library/service_settings_labels.py]: I actually changed it now to not be None but to have a default_factory=dict and that better fits the needs.

File: services/dask-sidecar/src/simcore_service_dask_sidecar/computational_sidecar/docker_utils.py (hunk 2 above)
Comment (sanderegg) [reply services/dask-sidecar/src/simcore_service_dask_sidecar/computational_sidecar/docker_utils.py:68]: was a copy/paste of code from before. but ok.

**Review by matusdrobuliak66 — APPROVED** [review]
Nice, thanks :+1:

**Pull request merged**